midp/gfx.js

```diff
@@ -1039,82 +996,24 @@ var currentlyFocusedTextEditor;
     }
 
     function clipRect(g, x, y, width, height) {
-        var translatedX1, translatedY1;
-        var translatedX2, translatedY2;
-
         if (width <= 0 || height <= 0) {
             g.clipX1 = g.clipY1 = g.clipX2 = g.clipY2 = 0;
-            g.clipped = true;
-            return;
-        }
-
-        // Translate the given coordinates
-        translatedX1 = x + g.transX;
-        translatedY1 = y + g.transY;
-
-        // Detect overflow
-        if (translatedX1 < 0) {
-            translatedX1 = (x < 0 || g.transX < 0) ? 0 : g.maxWidth;
-        }
-        if (translatedY1 < 0) {
-            translatedY1 = (y < 0 || g.transY < 0) ? 0 : g.maxHeight;
-        }
-
-        // If the passed in rect is below our current clip
-        if ((g.clipX2 < translatedX1) || (g.clipY2 < translatedY1)) {
-            // we have no intersection
-            g.clipX1 = g.clipY1 = g.clipX2 = g.clipY2 = 0;
-            g.clipped = true;
-            return;
-        }
-
-        if (translatedX1 > g.clipX1) {
-            g.clipX1 = (translatedX1 & 0x7fff);
-            g.clipped = true;
-        }
-
-        if (translatedY1 > g.clipY1) {
-            g.clipY1 = (translatedY1 & 0x7fff);
-            g.clipped = true;
-        }
-
-        // Start handling bottom right area
-
-        translatedX2 = x + g.transX + width;
-        translatedY2 = y + g.transY + height;
+        } else {
+            g.clipX1 = Math.max(g.clipX1, x + g.transX) & 0x7fff;
+            g.clipY1 = Math.max(g.clipY1, y + g.transY) & 0x7fff;
 
-        // Detect Overflow
-        if (translatedX2 < 0) {
-            translatedX2 = (x < 0 || g.transX < 0) ? translatedX1 : g.maxWidth;
-        }
-        if (translatedY2 < 0) {
-            translatedY2 = (y < 0 || g.transY < 0) ? translatedY1 : g.maxHeight;
```

Comment:
This section is also attempting to "detect overflow" in a way that doesn't make sense in JS.